staging/operator-lifecycle-manager/pkg/controller/operators/decorators/operator_test.go

```diff
@@ -230,3 +230,61 @@ func TestAddComponents(t *testing.T) {
 		})
 	}
 }
+
+func TestComponentLabelKey(t *testing.T) {
+	tests := []struct {
+		description       string
+		componentLabelKey string
+		name              string
+		expectedLabel     string
+		returnsError      bool
+	}{
+		{
+			description:       "when componentLabelKey is set then return it",
+			componentLabelKey: "my-component-label",
+			name:              "my-operator",
+			expectedLabel:     "my-component-label",
+			returnsError:      false,
+		},
+		{
+			description:       "when componentLabelKey is not set and operator name is less than 63 characters then do not truncate",
+			componentLabelKey: "",
+			name:              "my-operator",
+			expectedLabel:     ComponentLabelKeyPrefix + "my-operator",
+			returnsError:      false,
+		},
+		{
+			description:   "when componentLabelKey is not set and operator name is more than 63 characters truncate",
+			name:          "this-is-my-operator-its-the-coolest-you-got-a-problem-with-that-come-at-me-bro",
+			expectedLabel: ComponentLabelKeyPrefix + "this-is-my-operator-its-the-coolest-you-got-a-problem-with-that",
+			returnsError:  false,
+		},
+		{
+			description:   "when componentLabelKey is not set and operator name is more than 63 characters truncate and drop trailing illegal characters",
+			name:          "this-is-my-operator-its-the-coolest-you-got-a-problem-with----...---___...---",
+			expectedLabel: ComponentLabelKeyPrefix + "this-is-my-operator-its-the-coolest-you-got-a-problem-with",
+			returnsError:  false,
+		},
+		{
+			description:   "when componentLabelKey is not set and operator name is more than 63 characters and is made up of illegal characters then return error",
+			name:          "----...---___...-------...---___...-------...---___...-------...---___...---",
+			expectedLabel: "",
+			returnsError:  true,
+		},
+	}
```

Comment:
This doesn't have the case for a small label key with illegal characters...

Reply:
We'll need to fix this in master.